File: backend/models/contact_score.go
package models

// ContactScoreResponse is the API response for GET /contacts/:id/score
// (issue #383, ADR-0023) — the full explainable breakdown behind a
// contact's relationship health score, mirroring ContactBriefing's shape
// (one struct per facet, degrading independently) rather than leaking
// internal/scoring's Result type directly onto the wire (same reasoning as
// BriefingCadence wrapping services.CadenceHealth).
type ContactScoreResponse struct {
	ContactID uint `json:"contact_id"`
	// Score is 0-100; Band is one of moss/chanterelle/russula.
	Score int    `json:"score"`
	Band  string `json:"band"`

	Recency     ContactScoreFacet `json:"recency"`
	Frequency   ContactScoreFacet `json:"frequency"`
	Closeness   ContactScoreFacet `json:"closeness"`
	ReachOut    ContactScoreFacet `json:"reach_out"`
	LastUpdated ContactScoreFacet `json:"last_updated"`
}

// ContactScoreFacet is one weighted facet's contribution and plain-language
// explanation — the score is never a black box (issue #383 requirement 1).
type ContactScoreFacet struct {
	// Value is the facet's own 0-100 sub-score, before weighting.
	Value float64 `json:"value"`
	// Weight is the facet's share of the total (0-100; all five sum to 100).
	Weight float64 `json:"weight"`
	// Reason is a short, plain-language explanation of Value.
	Reason string `json:"reason"`
}

File: android/core/model/src/main/kotlin/com/mycorrhizal/crm/model/network/ContactScore.kt
package com.mycorrhizal.crm.model.network

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

// ---------------------------------------------------------------------------
// Relationship health score (issue #383, ADR-0023) — GET /contacts/:id/score
// ---------------------------------------------------------------------------

/**
 * `GET /contacts/:id/score` response — the full explainable breakdown behind
 * a contact's relationship health score, mirroring the backend's
 * `models.ContactScoreResponse` field-for-field (`backend/models/contact_score.go`).
 * [score] is 0-100; [band] is one of "moss"/"chanterelle"/"russula". The score
 * (and every facet sub-score) is computed server-side — Android renders it,
 * it is never recomputed on-device, the same read-only contract
 * [BriefingCadence]'s health carries.
 */
@JsonClass(generateAdapter = true)
data class ContactScoreResponse(
    @Json(name = "contact_id") val contactId: Int = 0,
    val score: Int = 0,
    val band: String = "",
    val recency: ContactScoreFacet = ContactScoreFacet(),
    val frequency: ContactScoreFacet = ContactScoreFacet(),
    val closeness: ContactScoreFacet = ContactScoreFacet(),
    @Json(name = "reach_out") val reachOut: ContactScoreFacet = ContactScoreFacet(),
    @Json(name = "last_updated") val lastUpdated: ContactScoreFacet = ContactScoreFacet(),
)

/**
 * One weighted facet's contribution and plain-language explanation — the
 * score is never a black box (backend `models.ContactScoreFacet`). [value] is
 * the facet's own 0-100 sub-score before weighting; [weight] is its share of
 * the total (all five facets sum to 100); [reason] is the short,
 * plain-language explanation shown in the facet breakdown.
 */
@JsonClass(generateAdapter = true)
data class ContactScoreFacet(
    val value: Double = 0.0,
    val weight: Double = 0.0,
    val reason: String = "",
)
